### PR TITLE
 Add CLI support for volume statedump

### PR DESCRIPTION
**Pull request by prashanthpai**
Usage:

```sh
$ glustercli volume statedump <volname>
$ glustercli volume statedump <volname> --quotad
$ glustercli volume statedump <volname> --client <ip>:<pid>
```

Updates #642

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -33,76 +33,7 @@
...
-    "wal",
-    "wal/walpb"
-  ]
+  packages = ["alarm","auth","auth/authpb","client","clientv3","clientv3/concurrency","clientv3/namespace","compactor","discovery","embed","error","etcdserver","etcdserver/api","etcdserver/api/etcdhttp","etcdserver/api/v2http","etcdserver/api/v2http/httptypes","etcdserver/api/v3client","etcdserver/api/v3election","etcdserver/api/v3election/v3electionpb","etcdserver/api/v3election/v3electionpb/gw","etcdserver/api/v3lock","etcdserver/api/v3lock/v3lockpb","etcdserver/api/v3lock/v3lockpb/gw","etcdserver/api/v3rpc","etcdserver/api/v3rpc/rpctypes","etcdserver/auth","etcdserver/etcdserverpb","etcdserver/etcdserverpb/gw","etcdserver/membership","etcdserver/stats","lease","lease/leasehttp","lease/leasepb","mvcc","mvcc/backend","mvcc/mvccpb","pkg/adt","pkg/contention","pkg/cors","pkg/cpuutil","pkg/crc","pkg/debugutil","pkg/fileutil","pkg/httputil","pkg/idutil","pkg/ioutil","pkg/logutil","pkg/monotime","pkg/netutil","pkg/pathutil","pkg/pbutil","pkg/runtime","pkg/schedule","pkg/srv","pkg/tlsutil","pkg/transport","pkg/types","pkg/wait","proxy/grpcproxy/adapter","raft","raft/raftpb","rafthttp","snap","snap/snappb","store","version","wal","wal/walpb"]
```
Comment (aravindavk) [inline Gopkg.lock]: I prefer the previous format, one package name in one line. Easy to recognize new dependency in patch diff

File: Gopkg.lock [hunk 2]
```diff
@@ -155,11 +86,7 @@
 
 [[projects]]
   name = "github.com/golang/protobuf"
-  packages = [
-    "jsonpb",
-    "proto",
-    "ptypes/struct"
-  ]
+  packages = ["jsonpb","proto","ptypes/struct"]
```
Comment (aravindavk) [inline Gopkg.lock]: same as above, patch diff will be readable in case of one package name per line

**Comment by aravindavk** [comment]
> Also bumped up cobra package to latest released version.

@kshlm do you see any issue with this dependency update?

File: Gopkg.lock (hunk 1 above)
Comment (prashanthpai) [reply Gopkg.lock]: I didn't do it - this is an auto-generated file. This happens because some of us are using different version of `dep`.

Comment (aravindavk) [reply Gopkg.lock]: Ok, I was confused with `Gopkg.toml` file

**Comment by prashanthpai** [comment]
retest this please

File: glustercli/cmd/volume-statedump.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+var (
+	flagStatedumpClientHost string
+	flagStatedumpVolumeName string
+	flagStatedumpClientPid  int
+)
+
+var volumeStatedumpCmd = &cobra.Command{
+	Use:   "statedump --volname <volname>",
```
Comment (kshlm) [inline glustercli/cmd/volume-statedump.go]: Doing it this way changes the well know volume command syntax of `gluster volume <sub-command> <volname> <other args or options>`.
The statedump command is still a volume subcommand, and should follow the others.

File: glustercli/cmd/volume-statedump.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	volumeCmd.AddCommand(volumeStatedumpCmd)
+}
+
+func volumeStatedumpCmdArgs(cmd *cobra.Command, args []string) error {
+
+	if len(args) != 1 {
+		return errors.New("need exactly one argument i.e name of the volume")
+	}
+
+	if _, err := cmd.Flags().GetBool("quota"); err != nil {
```
Comment (kshlm) [inline glustercli/cmd/volume-statedump.go]: You possibly don't need this, as pflag parsing should ensure that it is a bool, and flagStatedumpQuotad should be set correctly.

**Pull request merged**